assembly/src/ast/parsers/constants.rs

```diff
@@ -0,0 +1,262 @@
+use super::{Felt, LocalConstMap, ParsingError, StarkField, String, ToString, Token, Vec};
+use core::fmt::Display;
+
+// CONSTANT VALUE EXPRESSIONS
+// ================================================================================================
+
+/// An operation used in constant expressions
+#[derive(Debug, PartialEq)]
+enum Operation {
+    Add,
+    Sub,
+    Mul,
+    Div,
+    IntDiv,
+    LPar,
+    RPar,
+    Value(Felt),
+}
+
+impl Operation {
+    /// Returns operator from [Operation] based on provided character
+    pub fn get_operator(char: &char) -> Self {
+        match char {
+            '+' => Self::Add,
+            '-' => Self::Sub,
+            '*' => Self::Mul,
+            '/' => Self::Div,
+            '(' => Self::LPar,
+            ')' => Self::RPar,
+            _ => unreachable!("Invalid operator character"),
+        }
+    }
+}
+
+impl Display for Operation {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        use Operation::*;
+
+        match self {
+            Add => write!(f, "+"),
+            Sub => write!(f, "-"),
+            Mul => write!(f, "*"),
+            Div => write!(f, "/"),
+            IntDiv => write!(f, "//"),
+            LPar => write!(f, "("),
+            RPar => write!(f, ")"),
+            Value(v) => write!(f, "{}", v),
+        }
+    }
+}
+
+/// Calculates expression in the constant value.
+pub fn calculate_const_value(
+    op: &Token,
+    expression: &str,
+    constants: &LocalConstMap,
+) -> Result<Felt, ParsingError> {
+    let postfix_expression = build_postfix_expression(op, expression, constants)?;
+    evaluate_postfix_expression(op, expression, postfix_expression)
+}
+
+/// Parses constant value expression and transforms it into the postfix notation
+fn build_postfix_expression(
+    op: &Token,
+    expression: &str,
+    constants: &LocalConstMap,
+) -> Result<Vec<Operation>, ParsingError> {
+    let ops = ['+', '-', '*', '/', '(', ')'];
+
+    let mut stack = Vec::new();
+    let mut postfix_expression = Vec::new();
+    let mut parsed_value = String::new();
+
+    // Create a "window" of chars containing current and next char of the expression. It is needed
+    // to determine field division `//`.
+    let mut chars = expression.chars();
+    let mut next_char = chars.next();
+    let mut current_char: Option<char>;
+
+    loop {
+        current_char = next_char;
+        next_char = chars.next();
+
+        let current_char = current_char.unwrap();
```

Comment:
It seems like we have parsing and processing interconnected here. I wonder if it would make the code more clear if they were separate. Specifically, we could have some iterator which would have a `next()` method. This would return the next `Operation` and would do all string processing internally.